### ISSUE TITLE
better logs for ModelHubMixin unserialized parameters

### ISSUE THREAD

**Issue by not-lain** [huggingface/huggingface_hub#2297]
maybe add a warnings when instantiating a new model and when there have been problems with the serialization.
* input param of an unsupported  type => tell user to use the coders param and a link to docs
* when `param` exists in both `config` param as a key, and the init method has the same `param` name

tagging @Wauplin for better visibility

**Comment by Wauplin** [comment]
> example : remi's [repo](https://huggingface.co/cadene/2024_05_28_aloha_act_no_state_2_frames_aloha_v2_static_dora_test_wrist_gripper_040000/tree/main) on HF

What script/library is supposed to open this repo? And what type was not handled in the process?

**Comment by not-lain** [comment]
> What script/library is supposed to open this repo? And what type was not handled in the process?

this one: https://github.com/huggingface/lerobot/blob/d585c73f9f73ed3313f9e12969e60038389e6f59/lerobot/common/logger.py#L153

also, this one will give you a better look at the repo : https://github.com/huggingface/lerobot/blob/d585c73f9f73ed3313f9e12969e60038389e6f59/examples/2_evaluate_pretrained_policy.py#L24

for full transparency, they are using the ModelHubMixin only with the policies since they are using omegaconf, I believe they can optimize how they are loading and pushing their models by injecting the _cfg parameter into the policy init parameters making loading and pushing models is being done seamlessly with huggingface.

TLDR; 
they are using modelhubmixin with snapshotdownload, they are also working around the unsupported input type by detaching the OmegaConf parameters into a separate file.

In hindsight, this beats the purpose of using ModelHubMixin, but we are not here to enforce a specific architecture on them. 
well, this issue is an optional feature, but it would be nice to let developers know that their init parameters were not added properly to the _hub_mixin_config because of their type.

**Comment by not-lain** [comment]
> I would prefer not to. For ModelHubMixin, we should differenciate 2 types of users: the library maintainers that will integrate the mixin and the final users that will not know what to do with the warning.

reading this one I would agree with you, so i'm closing this one
